software/firmware/file_utils.py

```diff
@@ -0,0 +1,32 @@
+import os
```

Comment:
Nice! Thanks for abstracting this. This does a really good job of separating the reusable load/save with error checking behavior.